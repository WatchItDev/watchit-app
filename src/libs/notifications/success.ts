/**
 * Enum of our known success texts.
 */
export enum SUCCESS {
  FALLBACK = 'FALLBACK',
  DEPOSIT_SUCCESSFULLY = 'DEPOSIT_SUCCESSFULLY',
  WITHDRAW_SUCCESSFULLY = 'WITHDRAW_SUCCESSFULLY',
  INVITATIONS_SUCCESSFULLY = 'INVITATIONS_SUCCESSFULLY',
  LINK_COPIED_TO_CLIPBOARD = 'LINK_COPIED_TO_CLIPBOARD',
  WALLET_CHANGED_SUCCESFULLY = 'WALLET_CHANGED_SUCCESFULLY',
  PROFILE_METADATA_UPDATED = 'PROFILE_METADATA_UPDATED',
  JOINING_PRICE_SUCCESSFULLY = 'JOINING_PRICE_SUCCESSFULLY',
  PROFILE_UPDATED_SUCCESSFULLY = 'PROFILE_UPDATED_SUCCESSFULLY',
  PROFILE_CREATED_SUCCESSFULLY = 'PROFILE_CREATED_SUCCESSFULLY',
  TRANSFER_CREATED_SUCCESSFULLY = 'TRANSFER_CREATED_SUCCESSFULLY',
  TIP_CREATED_SUCCESSFULLY = 'TIP_CREATED_SUCCESSFULLY',
  FOLLOW_UNFOLLOW_SUCCESSFULLY = 'FOLLOW_UNFOLLOW_SUCCESSFULLY',
  PROFILE_JOINED_SUCCESSFULLY = 'PROFILE_JOINED_SUCCESSFULLY',
  OWNERSHIP_REGISTERED_SUCCESSFULLY = 'OWNERSHIP_REGISTERED_SUCCESSFULLY',

  // Metamask
  METAMASK_CONNECTED_SUCCESSFULLY = 'METAMASK_CONNECTED_SUCCESSFULLY',

  // Marketing
  STRATEGY_STORED_SUCCESSFULLY = 'STRATEGY_STORED_SUCCESSFULLY',
  CAMPAIGN_STORED_SUCCESSFULLY = 'CAMPAIGN_STORED_SUCCESSFULLY',
  CAMPAIGN_CREATED_SUCCESSFULLY = 'CAMPAIGN_CREATED_SUCCESSFULLY',
  CAMPAIGN_CONFIGURED_SUCCESSFULLY = 'CAMPAIGN_CONFIGURED_SUCCESSFULLY',
  CAMPAIGN_WITHDRAWN_SUCCESSFULLY = 'CAMPAIGN_WITHDRAWN_SUCCESSFULLY',
}

/**
 * Error messages dictionary mapping our ERROR_NAMES to user-facing strings.
 */
export const SUCCESS_MESSAGES: Record<SUCCESS, string> = {
  [SUCCESS.FALLBACK]: 'Fallback.',
  [SUCCESS.DEPOSIT_SUCCESSFULLY]: 'The deposit was successful.',
  [SUCCESS.WITHDRAW_SUCCESSFULLY]: 'The withdraw was successful.',
  [SUCCESS.INVITATIONS_SUCCESSFULLY]: 'The email invitation was sent successfully.',
  [SUCCESS.LINK_COPIED_TO_CLIPBOARD]: 'Link copied to clipboard!.',
  [SUCCESS.WALLET_CHANGED_SUCCESFULLY]: 'Wallet changed successfully!.',
  [SUCCESS.PROFILE_METADATA_UPDATED]: 'Profile metadata updated successfully.',
  [SUCCESS.JOINING_PRICE_SUCCESSFULLY]: 'Joining price set successfully!.',
  [SUCCESS.PROFILE_CREATED_SUCCESSFULLY]: 'Profile created successfully.',
  [SUCCESS.PROFILE_UPDATED_SUCCESSFULLY]: 'Profile updated successfully.',
  [SUCCESS.TRANSFER_CREATED_SUCCESSFULLY]: 'Transfer sent to {destination}.',
  [SUCCESS.TIP_CREATED_SUCCESSFULLY]: 'Thanks for your tip! Your support fuels more creations.',
  [SUCCESS.FOLLOW_UNFOLLOW_SUCCESSFULLY]: 'Successfully {profileName} {actionLbl}.',
  [SUCCESS.PROFILE_JOINED_SUCCESSFULLY]: 'Successfully joined the profile.',
  [SUCCESS.OWNERSHIP_REGISTERED_SUCCESSFULLY]: 'Asset {count} successfully processed',

  // Metamask
  [SUCCESS.METAMASK_CONNECTED_SUCCESSFULLY]: 'MetaMask connected successfully!',

  // Marketing
  [SUCCESS.STRATEGY_STORED_SUCCESSFULLY]: 'Strategy stored successfully!',
  [SUCCESS.CAMPAIGN_STORED_SUCCESSFULLY]: 'Campaign stored successfully!',
  [SUCCESS.CAMPAIGN_CREATED_SUCCESSFULLY]: 'Campaign created successfully!',
  [SUCCESS.CAMPAIGN_CONFIGURED_SUCCESSFULLY]: 'Campaign configured successfully!',
  [SUCCESS.CAMPAIGN_WITHDRAWN_SUCCESSFULLY]: 'Funds withdrawn from campaign successfully!',
};
