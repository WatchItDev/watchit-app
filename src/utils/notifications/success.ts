/**
 * Enum of our known success texts.
 */
export enum SUCCESS {
  FALLBACK = "FALLBACK",
  DEPOSIT_SUCCESSFULLY = "DEPOSIT_SUCCESSFULLY",
  WITHDRAW_SUCCESSFULLY = "WITHDRAW_SUCCESSFULLY",
  INVITATIONS_SUCCESSFULLY = "INVITATIONS_SUCCESSFULLY",
  LINK_COPIED_TO_CLIPBOARD = "LINK_COPIED_TO_CLIPBOARD",
  WALLET_CHANGED_SUCCESFULLY = "WALLET_CHANGED_SUCCESFULLY",
  PROFILE_METADATA_UPDATED = "PROFILE_METADATA_UPDATED",
  JOINING_PRICE_SUCCESSFULLY = "JOINING_PRICE_SUCCESSFULLY",
  PROFILE_UPDATED_SUCCESSFULLY = "PROFILE_UPDATED_SUCCESSFULLY",
  PROFILE_CREATED_SUCCESSFULLY = "PROFILE_CREATED_SUCCESSFULLY",
  TRANSFER_CREATED_SUCCESSFULLY = "TRANSFER_CREATED_SUCCESSFULLY",
  FOLLOW_UNFOLLOW_SUCCESSFULLY = "FOLLOW_UNFOLLOW_SUCCESSFULLY",
  PROFILE_JOINED_SUCCESSFULLY = "PROFILE_JOINED_SUCCESSFULLY",
  OWNERSHIP_REGISTERED_SUCCESSFULLY = "OWNERSHIP_REGISTERED_SUCCESSFULLY",

  // Metamask
  METAMASK_CONNECTED_SUCCESSFULLY = "METAMASK_CONNECTED_SUCCESSFULLY",
}

/**
 * Error messages dictionary mapping our ERROR_NAMES to user-facing strings.
 */
export const SUCCESS_MESSAGES: Record<SUCCESS, string> = {
  [SUCCESS.FALLBACK]: "Fallback.",
  [SUCCESS.DEPOSIT_SUCCESSFULLY]: "The deposit was successful.",
  [SUCCESS.WITHDRAW_SUCCESSFULLY]: "The withdraw was successful.",
  [SUCCESS.INVITATIONS_SUCCESSFULLY]: "The email invitation was sent successfully.",
  [SUCCESS.LINK_COPIED_TO_CLIPBOARD]: "Link copied to clipboard!.",
  [SUCCESS.WALLET_CHANGED_SUCCESFULLY]: "Wallet changed successfully!.",
  [SUCCESS.PROFILE_METADATA_UPDATED]: "Profile metadata updated successfully.",
  [SUCCESS.JOINING_PRICE_SUCCESSFULLY]: "Joining price set successfully!.",
  [SUCCESS.PROFILE_CREATED_SUCCESSFULLY]: "Profile created successfully.",
  [SUCCESS.PROFILE_UPDATED_SUCCESSFULLY]: "Profile updated successfully.",
  [SUCCESS.TRANSFER_CREATED_SUCCESSFULLY]: "Transfer sent to {destination}.",
  [SUCCESS.FOLLOW_UNFOLLOW_SUCCESSFULLY]: "Successfully {profileName} {actionLbl}.",
  [SUCCESS.PROFILE_JOINED_SUCCESSFULLY]: "Successfully joined the profile.",
  [SUCCESS.OWNERSHIP_REGISTERED_SUCCESSFULLY]: "Asset {count} successfully processed",

  // Metamask
  [SUCCESS.METAMASK_CONNECTED_SUCCESSFULLY]: "MetaMask connected successfully!",
};
