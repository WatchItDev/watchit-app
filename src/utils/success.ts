/**
 * Enum of our known error names.
 */
export enum SUCCESS {
  DEPOSIT_SUCCESSFULLY = 'DEPOSIT_SUCCESSFULLY',
}

/**
 * Error messages dictionary mapping our ERROR_NAMES to user-facing strings.
 */
export const SUCCESS_MESSAGES: Record<SUCCESS, string> = {
  [SUCCESS.DEPOSIT_SUCCESSFULLY]: 'The deposit was successful.',
};
