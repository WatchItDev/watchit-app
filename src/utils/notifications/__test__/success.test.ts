import { describe, it, expect } from 'vitest';
import { SUCCESS, SUCCESS_MESSAGES } from '../success';

describe('[UTILS]: Success', () => {
  describe('SUCCESS_MESSAGES', () => {
    const containsPlaceholders = (message: string): boolean =>
      message.includes('{') && message.includes('}');

    it('should match the snapshot of all success messages', () => {
      expect(SUCCESS_MESSAGES).toMatchSnapshot();
    });

    it('should have a message for every success type', () => {
      Object.values(SUCCESS).forEach(successKey => {
        expect(SUCCESS_MESSAGES[successKey as SUCCESS]).toBeDefined();
        expect(typeof SUCCESS_MESSAGES[successKey as SUCCESS]).toBe('string');
        expect(SUCCESS_MESSAGES[successKey as SUCCESS].length).toBeGreaterThan(0);
      });
    });

    it('should not have extra messages that do not correspond to success types', () => {
      Object.keys(SUCCESS_MESSAGES).forEach(messageKey => {
        expect(Object.values(SUCCESS)).toContain(messageKey);
      });
    });

    it('should have the correct message for FALLBACK success', () => {
      expect(SUCCESS_MESSAGES[SUCCESS.FALLBACK]).toBe('Fallback.');
    });

    it('should have the correct message for DEPOSIT_SUCCESSFULLY success', () => {
      expect(SUCCESS_MESSAGES[SUCCESS.DEPOSIT_SUCCESSFULLY]).toBe('The deposit was successful.');
    });

    it('should contain placeholders in specific success messages', () => {
      expect(containsPlaceholders(SUCCESS_MESSAGES[SUCCESS.TRANSFER_CREATED_SUCCESSFULLY])).toBe(true);
      expect(containsPlaceholders(SUCCESS_MESSAGES[SUCCESS.FOLLOW_UNFOLLOW_SUCCESSFULLY])).toBe(true);
      expect(containsPlaceholders(SUCCESS_MESSAGES[SUCCESS.OWNERSHIP_REGISTERED_SUCCESSFULLY])).toBe(true);
    });

    it('should have the correct placeholder format in messages', () => {
      const messagesWithPlaceholders = Object.values(SUCCESS_MESSAGES).filter(containsPlaceholders);

      messagesWithPlaceholders.forEach(message => {
        const placeholderRegex = /{[a-zA-Z0-9]+}/g;
        const matches = message.match(placeholderRegex) || [];

        expect(matches.length).toBeGreaterThan(0);
        matches.forEach(match => {
          expect(match.startsWith('{')).toBe(true);
          expect(match.endsWith('}')).toBe(true);
          expect(match.length).toBeGreaterThan(2);
        });
      });
    });

    it('should have consistent transaction-related success messages', () => {
      const transactionSuccesses = [
        SUCCESS.DEPOSIT_SUCCESSFULLY,
        SUCCESS.WITHDRAW_SUCCESSFULLY,
      ];

      transactionSuccesses.forEach(success => {
        expect(SUCCESS_MESSAGES[success]).toBeDefined();
        expect(SUCCESS_MESSAGES[success].toLowerCase()).toMatch(/successful|was successful/);
      });
    });

    it('should have consistent profile-related success messages', () => {
      const profileSuccesses = [
        SUCCESS.PROFILE_METADATA_UPDATED,
        SUCCESS.PROFILE_CREATED_SUCCESSFULLY,
        SUCCESS.PROFILE_UPDATED_SUCCESSFULLY,
        SUCCESS.PROFILE_JOINED_SUCCESSFULLY,
      ];

      profileSuccesses.forEach(success => {
        expect(SUCCESS_MESSAGES[success]).toBeDefined();
        expect(SUCCESS_MESSAGES[success].toLowerCase()).toMatch(/profile|successfully/);
      });
    });

    it('should have consistent marketing-related success messages', () => {
      const marketingSuccesses = [
        SUCCESS.STRATEGY_STORED_SUCCESSFULLY,
        SUCCESS.CAMPAIGN_STORED_SUCCESSFULLY,
        SUCCESS.CAMPAIGN_CREATED_SUCCESSFULLY,
        SUCCESS.CAMPAIGN_CONFIGURED_SUCCESSFULLY,
        SUCCESS.CAMPAIGN_WITHDRAWN_SUCCESSFULLY,
      ];

      marketingSuccesses.forEach(success => {
        expect(SUCCESS_MESSAGES[success]).toBeDefined();
        expect(SUCCESS_MESSAGES[success].toLowerCase()).toMatch(/campaign|strategy|successfully/);
      });
    });
  });
});
