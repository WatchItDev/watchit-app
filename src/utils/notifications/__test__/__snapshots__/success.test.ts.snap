// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[UTILS]: Success > SUCCESS_MESSAGES > should match the snapshot of all success messages 1`] = `
{
  "CAMPAIGN_CONFIGURED_SUCCESSFULLY": "Campaign configured successfully!",
  "CAMPAIGN_CREATED_SUCCESSFULLY": "Campaign created successfully!",
  "CAMPAIGN_STORED_SUCCESSFULLY": "Campaign stored successfully!",
  "CAMPAIGN_WITHDRAWN_SUCCESSFULLY": "Funds withdrawn from campaign successfully!",
  "DEPOSIT_SUCCESSFULLY": "The deposit was successful.",
  "FALLBACK": "Fallback.",
  "FOLLOW_UNFOLLOW_SUCCESSFULLY": "Successfully {profileName} {actionLbl}.",
  "INVITATIONS_SUCCESSFULLY": "The email invitation was sent successfully.",
  "JOINING_PRICE_SUCCESSFULLY": "Joining price set successfully!.",
  "LINK_COPIED_TO_CLIPBOARD": "Link copied to clipboard!.",
  "METAMASK_CONNECTED_SUCCESSFULLY": "MetaMask connected successfully!",
  "OWNERSHIP_REGISTERED_SUCCESSFULLY": "Asset {count} successfully processed",
  "PROFILE_CREATED_SUCCESSFULLY": "Profile created successfully.",
  "PROFILE_JOINED_SUCCESSFULLY": "Successfully joined the profile.",
  "PROFILE_METADATA_UPDATED": "Profile metadata updated successfully.",
  "PROFILE_UPDATED_SUCCESSFULLY": "Profile updated successfully.",
  "STRATEGY_STORED_SUCCESSFULLY": "Strategy stored successfully!",
  "TRANSFER_CREATED_SUCCESSFULLY": "Transfer sent to {destination}.",
  "WALLET_CHANGED_SUCCESFULLY": "Wallet changed successfully!.",
  "WITHDRAW_SUCCESSFULLY": "The withdraw was successful.",
}
`;
